cdk/domino_cdk/provisioners/vpc.py

```diff
@@ -170,12 +173,16 @@ def provision_vpc(self, stack_name: str, vpc: config.VPC, monitoring_bucket: Opt
     def provision_bastion(self, name: str, bastion: config.VPC.Bastion) -> Optional[ec2.SecurityGroup]:
         if not bastion.enabled:
             return None
+
+        root_device_name = "/dev/xvda"  # This only works for AL2
         if bastion.ami_id:
             region = cdk.Stack.of(self.scope).region
             machine_image = ec2.MachineImage.generic_linux(
                 {region: bastion.ami_id},
-                user_data=ec2.UserData.custom(bastion.user_data),
+                user_data=ec2.UserData.custom(bastion.user_data) if bastion.user_data else None,
```

Comment:
Hmmm. So it did not work with AMI without user_data

Now we can specify ami_id without user_data. 

It is the change we were talking about but it strange to see it only in this place.

Reply:
Well, this is the bastion, so there's nothing "running" on it necessarily. I think it's ok for the bastion to have optional user data for a custom AMI.

Reply:
AFAIR, if we specify custom AMI, 

```
        user_data: ... - Custom script for user_data, ran by cloud_init on node startup. When using the
                         default EKS AMI, this does not replace the default user_data. Your custom script
                         will be injected _before_ the default one, and both will be ran. However, when
                         specifying a custom AMI, this will be the *only* user_data script in use.
```

Does it work?

Can we use custom AMI and not to specify user data and still get something running? 

I did such experiment but I do not remember the outcome already.

Reply:
Sorry, this change is only for the bastion which doesn't need to run anything from user-data to function correctly. For EKS nodegroup nodes you're definitely correct, we should require user-data.